src/lavinmq/shovel/shovel.cr

```diff
@@ -41,7 +41,8 @@ module LavinMQ
 
       def initialize(@name : String, @uri : URI, @queue : String?, @exchange : String? = nil,
                      @exchange_key : String? = nil,
-                     @delete_after = DEFAULT_DELETE_AFTER, @prefetch = DEFAULT_PREFETCH, @ack_mode = DEFAULT_ACK_MODE,
+                     @delete_after = DEFAULT_DELETE_AFTER, @prefetch = DEFAULT_PREFETCH,
+                     @ack_mode = DEFAULT_ACK_MODE, @consumer_args = {} of String => JSON::Any || nil,
```

Comment:
Any reason to have `@consumer_args` as a instance variable? Looks like it's only used to populate `@args`?

And the `|| nil` part seems odd. Will it ever do anything? Default value is empty hash. Or do you want `@consumer_args : Hash(String, JSON::Any)? = nil`?

Reply:
Yeah, looks much better that way 👍